app_server/app/services/StreamService.js

```diff
@@ -59,17 +59,75 @@ Class.getStreamById = function(streamId) {
 Class.getListOfStreams = function(filters) {
   logger.debug('Getting list of streams with filters: %j', filters);
 
-  return Storage.getListOfStreams(filters).then(function receiveResult(result) {
-    if (result) {
-      return result.map(Utility.formatViewObject);
-    } else {
-      return Promise.resolve(new CustomError
-        .NotFoundError('Stream not found'));
-    }
-  }).catch(function(err) {
-    logger.error(err);
-    return null;
-  });
+  return Storage.getListOfStreams(filters)
+    .then(function receiveResult(results) {
+      if (results) {
+        return results.map((singleStream) =>
+          Utility.formatStreamObject(singleStream, 'view'));
+      } else {
+        return Promise.resolve(new CustomError
+          .NotFoundError('Stream not found'));
+      }
```

Comment:
Just return `new CustomError...` directly. I think the `Promise.resolve` is not needed as `bluebird` can figure out what to do either when a raw value or a `Promise` is returned.